contracts/0.8.9/SelfOwnedStETHBurner.sol

```diff
@@ -0,0 +1,312 @@
+        uint256 amount,
+        uint256 sharesAmount
+    );
+
+    /**
+      * Emitted when the stETH `amount` (corresponding to `sharesAmount` shares) burnt for the `isCover` reason.
+      */
+    event StETHBurnt(
+        bool indexed isCover,
+        uint256 amount,
+        uint256 sharesAmount
+    );
+
+    /**
+      * Emitted when the excessive stETH `amount` (corresponding to `sharesAmount` shares) recovered (e.g. transferred)
+      * to the Lido treasure address by `requestedBy` sender.
+      */
+    event ExcessStETHRecovered(
+        address indexed requestedBy,
+        uint256 amount,
+        uint256 sharesAmount
+    );
+
+    /**
+      * Emitted when the ERC20 `token` recovered (e.g. transferred)
+      * to the Lido treasure address by `requestedBy` sender.
+      */
+    event ERC20Recovered(
+        address indexed requestedBy,
+        address indexed token,
+        uint256 amount
+    );
+
+    /**
+      * Emitted when the ERC721-compatible `token` (NFT) recovered (e.g. transferred)
+      * to the Lido treasure address by `requestedBy` sender.
+      */
+    event ERC721Recovered(
+        address indexed requestedBy,
+        address indexed token,
+        uint256 tokenId
+    );
+
+    /**
+      * Ctor
+      *
+      * @param _treasury the Lido treasury address (see StETH/ERC20/ERC721-recovery interfaces)
+      * @param _lido the Lido token (stETH) address
+      * @param _voting the Lido Aragon Voting address
+      * @param _totalCoverSharesBurnt Shares burnt counter init value (cover case)
+      * @param _totalNonCoverSharesBurnt Shares burnt counter init value (non-cover case)
+      */
+    constructor(
+        address _treasury,
+        address _lido,
+        address _voting,
+        uint256 _totalCoverSharesBurnt,
+        uint256 _totalNonCoverSharesBurnt
+    ) {
+        require(_treasury != address(0), "TREASURY_ZERO_ADDRESS");
+        require(_lido != address(0), "LIDO_ZERO_ADDRESS");
+        require(_voting != address(0), "VOTING_ZERO_ADDRESS");
+
+        TREASURY = _treasury;
+        LIDO = _lido;
+        VOTING = _voting;
+
+        totalCoverSharesBurnt = _totalCoverSharesBurnt;
+        totalNonCoverSharesBurnt = _totalNonCoverSharesBurnt;
+    }
+
+    /**
+      * @notice BE CAREFUL, the provided stETH will be burnt permanently.
+      *
+      * Transfers `_stETH2Burn` stETH tokens from the message sender and irreversibly locks these
+      * on the burner contract address. Internally converts `_stETH2Burn` amount into underlying
+      * shares amount (`_stETH2BurnAsShares`) and marks the converted amount for burning
+      * by increasing the `coverSharesBurnRequested` counter.
+      *
+      * @param _stETH2Burn stETH tokens to burn
+      */
+    function requestBurnMyStETHForCover(uint256 _stETH2Burn) external {
+        _requestBurnMyStETH(_stETH2Burn, true);
+    }
+
+    /**
+      * @notice BE CAREFUL, the provided stETH will be burnt permanently.
```

Comment:
It's worth to add to the doc that only `voting` could burn the stETH.

Reply:
Because the check `require(msg.sender == VOTING, "MSG_SENDER_MUST_BE_VOTING");` is in the nested call it's not obvious from the first glance

Reply:
+